clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -5612,6 +5612,114 @@ test_unscheduled_initially_roi()
 #endif
 }
 
+static void
+test_unscheduled_initially_rebalance()
+{
+    // Tests i#7318 where on a rebalance a too-large runqueue has nothing
+    // but blocked inputs. That's easiest to make happen at the init-time
+    // rebalance where we create a bunch of starts-unscheduled (but not infinite)
+    // inputs.
+    std::cerr << "\n----------------\nTesting initially-unscheduled init rebalance\n";
+    static constexpr int NUM_OUTPUTS = 3;
+    static constexpr int NUM_INPUTS = 5;
+    static constexpr int BLOCK_LATENCY = 100;
+    static constexpr double BLOCK_SCALE = 1. / (BLOCK_LATENCY);
+    static constexpr uint64_t BLOCK_TIME_MAX = 500;
+    static constexpr int MIGRATION_THRESHOLD = 0;
+    static constexpr memref_tid_t TID_BASE = 100;
+    std::vector<trace_entry_t> refs[NUM_INPUTS];
+    for (int i = 0; i < NUM_INPUTS; ++i) {
+        if (i == NUM_OUTPUTS - 1) {
```

Comment:
Does this deliberately use a "middle" thread? Or else I'd've expected just a `i==0` or `i==NUM_INPUTS-1`.

Reply:
It did deliberately want the last output: that was my initial attempt to reproduce this.  With the bindings approach that's no longer needed; changed to i==0.

Reply:
Not sure I like i==0: it now successfully moves the live input; it then tries to move more and fails later.  Changed it to i==1 which is a more straightforward to understand scenario.

Reply:
Grrr, that was w/o the binding: the binding applies to the live too.  Could split the workload and not apply the binding to the live.  Or could live w/ the live having the binding and being moved and reproducing on the 2nd move attempt; it is unbalanced enough for that.  Going with that and back to i==0.